soda_machine.rb

```diff
@@ -7,12 +7,17 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.find { |soda| soda.brand == soda_brand }
   end
 
   def sell(soda_brand)
+    sold_soda = find_soda(soda_brand)
+    @cash += sold_soda.price if sold_soda
+    @sodas.delete(sold_soda)
```

Comment:
Thanks James - Just to be clear, you're suggesting that the method would still run if I just chopped the `if sold_soda` at the end of line 19? I believe I was fidgeting with a test to get the desired return value of nil if the soda didn't exist. But if line 18 evaluates to nil at the outset, then the next two lines would just fail silently and the method would return nil?